src/MCEventsProcessing/MCEventsProcessing.jl

```diff
@@ -76,6 +76,10 @@ function simulate_waveforms( mcevents_table::AbstractVector{<:NamedTuple}, sim::
                              geometry_check::Bool = false,
                              verbose::Bool = false ) where {T <: SSDFloat}
     mcevents = TypedTables.Table(mcevents_table)
+    #ToDo: Provisional until we support units in CartesianPoints
+    unitless_pos = [collect(to_internal_units.(v)) for v in mcevents.pos]
+    mcevents = TypedTables.Table((evtno = mcevents.evtno, detno = mcevents.detno, thit  = mcevents.thit, edep  = mcevents.edep, pos   = unitless_pos))
+
```

Comment:
I'm not a fan of this, as it changes the `pos` column of the `Table` as output of this function. I can also see that we have a `_get_unitless_position` function beneath this function, can we not use this WITHIN this function, but still return the OLD table with the new column with waveforms at the end of this function?